src/ParallelAlgorithms/Actions/SetData.hpp

```diff
@@ -0,0 +1,53 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/DataBox/DataBoxTag.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace Parallel {
+template <typename Metavariables>
+struct ConstGlobalCache;
+}  // namespace Parallel
+/// \endcond
+
+namespace Actions {
+
+/*!
+ * \ingroup ActionsGroup
+ * \brief Mutate the DataBox tags in `TagsList` according to the `data`.
+ *
+ * DataBox changes:
+ * - Modifies:
+ *   - All tags in `TagsList`
+ */
+template <typename TagsList>
+struct SetData;
+
+template <typename... Tags>
+struct SetData<tmpl::list<Tags...>> {
+  template <
```

Comment:
Nitpick, but did you clang format? Normally I thought clang-format doesn't leave spaces like this unless the <> aren't balanced

Reply:
yea, this is what clang-format wants :)